datafusion/common/src/scalar.rs

```diff
@@ -1391,6 +1391,29 @@ impl ScalarValue {
                 }
                 Self::Struct(Some(Box::new(field_values)), Box::new(fields.clone()))
             }
+            DataType::FixedSizeList(nested_type, _len) => {
+                let list_array = array
+                    .as_any()
+                    .downcast_ref::<FixedSizeListArray>()
+                    .ok_or_else(|| {
+                        DataFusionError::Internal(
+                            "Failed to downcast FixedSizeListArray".to_string(),
+                        )
+                    })?;
```

Comment:
This is fine -- I also think it is fine to panic in this case where the data type didn't match the array type as this is not an error condition DataFusion is expected to handle in the normal case:

```suggestion
                    .unwrap();
```